gcloud-java-core/src/main/java/com/google/gcloud/AuthCredentials.java

```diff
@@ -42,8 +38,66 @@
   private static class AppEngineAuthCredentials extends AuthCredentials {
 
     private static final AuthCredentials INSTANCE = new AppEngineAuthCredentials();
-    private static final AppEngineAuthCredentialsState STATE =
-        new AppEngineAuthCredentialsState();
+    private static final AppEngineAuthCredentialsState STATE = new AppEngineAuthCredentialsState();
+
+    private static class AppEngineCredentials extends GoogleCredentials {
+
+      private final Object appIdentityService;
+      private final Collection<String> scopes;
+      private final boolean scopesRequired;
+
+      AppEngineCredentials() {
+        try {
+          Class<?> factoryClass =
+              Class.forName("com.google.appengine.api.appidentity.AppIdentityServiceFactory");
```

Comment:
I want you to keep a reference here for `appIdentityService` and `getAccessTokenResult` and `getAccessToken`.